src/common/tags.c

```diff
@@ -214,7 +214,7 @@ gboolean dt_tag_new(const char *name, guint *tagid)
     // tagid already exists.
     if(tagid != NULL) *tagid = sqlite3_column_int64(stmt, 0);
     sqlite3_finalize(stmt);
-    return TRUE;
+    return FALSE;
```

Comment:
This is not correct. Callers expect TRUE is tag exists or is created.

Reply:
in tag.c line 248 dt_tag_new answer is used to raise signal.
But development doesn't create new tag. It only attaches "changed" tag. So this shouldn't impact the response time.

Reply:
But what the point of raising a signal if the tag was already present ? The comment says:

`  /* if everything went fine, raise signal of tags change to refresh keywords module in GUI */`

But what to refresh is nothing has changed ?

Reply:
The fact is that the tag already exists is not an issue. 
I agree it is useless to refresh if nothing has changed. But this should not occur that often.
Only lua calls dt_tag_new_from_gui() (and makes the signal). This is not related to the slowness issue. Development calls dt_tag_new().
I think your other change, on attachment ("changed") will make the difference.

Reply:
I agree this is not the area where I gain the speed, but yet this seems quite superfluous. I'll remove this and will merge.